wrappers/rust/icicle-cuda-runtime/src/memory.rs

```diff
@@ -117,6 +134,66 @@ impl<T> HostSlice<T> {
         self.0
             .iter_mut()
     }
+
+    pub fn is_pinnable(&self) -> bool {
+        let pinnable = get_device_attribute(cudaDeviceAttr::cudaDevAttrHostRegisterSupported, 0).unwrap();
```

Comment:
why hardcoded device 0?

Reply:
Removed for now as there were issues with using cudaHostRegister. But in general, you're correct, theres no reason to specifically check device 0 and this should be more robust to ensure some device supports cudaHostRegister